packages/x-data-grid/src/hooks/utils/useGridSelector.ts

```diff
@@ -22,6 +34,21 @@ function applySelector<Api extends GridApiCommon, T>(
   return selector(apiRef.current.state);
 }
 
+function applySelectorV8<Api extends GridApiCommon, Args, T>(
+  apiRef: React.MutableRefObject<Api>,
+  selector: Selector<Api, Args, T>,
+  args: Args,
+  instanceId: GridCoreApi['instanceId'],
+) {
+  if (isOutputSelector(selector)) {
+    if (isArgumentsSelector(selector)) {
+      return selector(apiRef, args);
+    }
+    return selector(apiRef);
```

Comment:
Is this logic necessary? Can't we just `return selector(apiRef, args)` for both cases?

Reply:
Agreed, we could certainly simplify it. 
Updated: https://github.com/mui/mui-x/pull/14236/commits/97348c2b29c6078637ab2df32c347a1ff54b7803